server/src/main/java/io/spine/server/ServerEnvironment.java

```diff
@@ -198,109 +196,127 @@ public void configureDeployment(Supplier<DeploymentType> supplier) {
     }
 
     /**
-     * Assigns {@code StorageFactory} for the production mode of the application.
+     * Assigns the specified {@code TransportFactory} for the specified application environment.
      *
-     * @see #configureStorageForTests(StorageFactory)
+     * @return this instance of {@code ServerEnvironment}
      */
-    public void configureStorage(StorageFactory productionStorageFactory) {
-        checkNotNull(productionStorageFactory);
-        this.productionStorageFactory = wrap(productionStorageFactory);
+    @CanIgnoreReturnValue
+    public ServerEnvironment use(StorageFactory storage, EnvironmentType envType) {
+        checkNotNull(envType);
+        environment().register(envType);
+        this.storageFactory.use(wrap(storage), envType);
+        return this;
     }
```

Comment:
You could extract the common part from each of these methods (and check for `null` the remaining parameter):
```java
    private static <V> void use(V value, EnvSetting<V> setting, EnvironmentType type) {
        checkNotNull(value);
        checkNotNull(type);
        checkNotNull(setting);
        environment().register(type);
        setting.use(value, type);
    }
```

The public methods will then look like this:
```java
    @CanIgnoreReturnValue
    public ServerEnvironment use(StorageFactory factory, EnvironmentType envType) {
        SystemAwareStorageFactory wrapped = wrap(factory);
        use(wrapped, this.storageFactory, envType);
        return this;
    }
```